src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -247,4 +248,40 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
   return result;
 }
 
+int verify_crl_signature(X509_CRL* crl, X509* issuer) {
+  EVP_PKEY* ikey = X509_get_pubkey(issuer);
+  if (ikey == nullptr) {
+    // Can't verify signature because we couldn't get the pubkey, fail the
+    // check.
+    EVP_PKEY_free(ikey);
+    return 1;
+  }
+  int ret = X509_CRL_verify(crl, ikey);
+  EVP_PKEY_free(ikey);
+  return ret;
+}
+
+int verify_crl_cert_issuer_names_match(X509_CRL* crl, X509* issuer) {
+  return X509_NAME_cmp(X509_get_issuer_name(issuer), X509_CRL_get_issuer(crl));
+}
+
+bool verify_crl_sign_bit(X509* issuer) {
+  return (X509_get_key_usage(issuer) & KU_CRL_SIGN) != 0;
+}
+
+std::string IssuerFromCert(X509* cert) {
+  if (cert == nullptr) {
+    return "";
+  }
+  X509_NAME* issuer = X509_get_issuer_name(cert);
+  int len;
```

Comment:
Any reason this is broken up? I would have expected this to be written as:
```
unsigned char* buf = nullptr;
int len = i2d_X509_NAME(issuer, &buf);
```

Reply:
Again, copying from OpenSSL reference will clean up

Reply:
Pinging this, since it seems to still be the case.

Reply:
I think this comment is on an old commit, I've fixed it on the tip of this branch

Reply:
It looks like there are a couple instances where this still occurs, I'll comment on them.